src/Microsoft.Performance.SDK.Runtime/VersionChecker.cs

```diff
@@ -0,0 +1,153 @@
+        /// </summary>
+        /// <returns>
+        ///     A new instance of the <see cref="VersionChecker"/> class.
+        /// </returns>
+        public static VersionChecker Create()
+        {
+            return new VersionCheckerImpl();
+        }
+
+        /// <summary>
+        ///     Determines whether the given Version of the SDK is
+        ///     supported by the host.
+        /// </summary>
+        /// <param name="candidateVersion">
+        ///     The version of the SDK being used by the Plugin.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c> if the Plugin's version of the SDK is
+        ///     compatible with this version of the SDK; <c>false</c>
+        ///     otherwise.
+        /// </returns>
+        public abstract bool IsVersionSupported(SemanticVersion candidateVersion);
+
+        /// <summary>
+        ///     Determines if the given <see cref="Assembly"/> references
+        ///     the SDK, and if so, returns the <see cref="SemanticVersion"/> of the
+        ///     referenced SDK. If the given Assembly does not reference
+        ///     the SDK, then <c>null</c> is returned.
+        /// </summary>
+        /// <param name="candidate">
+        ///     The <see cref="Assembly"/> to check for a referenced SDK.
+        /// </param>
+        /// <returns>
+        ///     The <see cref="SemanticVersion"/> of the referenced SDK, if it
+        ///     exists; <c>null</c> if the SDK is not referenced.
+        /// </returns>
+        public abstract SemanticVersion FindReferencedSdkVersion(Assembly candidate);
+
+        private sealed class VersionCheckerImpl
+            : VersionChecker
+        {
+            /// <summary>
+            ///     Gets the Version of the SDK that is being hosted.
+            /// </summary>
+            public override SemanticVersion Sdk => SdkVersion;
+
+            /// <summary>
+            ///     Gets the minimum supported Version of the SDK.
+            /// </summary>
+            public override SemanticVersion Baseline => BaselineVersion;
+
+            /// <summary>
+            ///     Determines whether the given Version of the SDK is
+            ///     supported by the host.
+            /// </summary>
+            /// <param name="candidateVersion">
+            ///     The version of the SDK being used by the Plugin.
+            /// </param>
+            /// <returns>
+            ///     <c>true</c> if the Plugin's version of the SDK is
+            ///     compatible with this version of the SDK; <c>false</c>
+            ///     otherwise.
+            /// </returns>
+            public override bool IsVersionSupported(
+                SemanticVersion candidateVersion)
+            {
+                Guard.NotNull(candidateVersion, nameof(candidateVersion));
+
+                return
+                    candidateVersion >= this.Baseline &&
+                    this.Sdk.Major == candidateVersion.Major;
```

Comment:
Is this right? Consider if `Baseline = 1.1.0` and `candidateVersion = 1.2.0`

Since candidate uses `1.2.0`, it potentially has calls to methods that don't exist in `1.1.0`. It's not a breaking change to add new methods, so it would be a minor bump. But that call isn't compatible.

I think this should be:

```cs
if (this.Sdk.Major != candidateVersion.Major)
{
    return false;
}

if (this.Sdk.Major == 0)
{
    // pre-release case: anything between baseline and current version is compatible
    // anything over current version is not guaranteed
    return candidateVersion >= this.Baseline && candidateVersion <= this.Sdk;
}
else
{
    // Older versions of same major release are compatible with this version
    return candidateVersion <= this.Baseline;
}
```

Reply:
Yes, that is correct. Thanks for catching that!